src/Compiler/CompileExpr.idr

```diff
@@ -208,6 +209,10 @@ builtinMagic = do
         , MagicCCon cons.succ 1
              (\ fc, [k] => CApp fc (CRef fc (UN "prim__add_Integer")) [CPrimVal fc (BI 1), k])
         ] -- TODO: add builtin pragmas for Nat related functions (to/from Integer, add, mult, minus, compare)
+    natToIntMagic : (Name, NatToInt) -> Magic
+    natToIntMagic (fn, MkNatToInt {arity, natIdx}) =
```

Comment:
Please don't use named pattern-matching in the compiler: the syntax will probably have
to change in a backwards incompatible way in the future because of #965.